src/App.tsx

```diff
@@ -8,7 +8,49 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [areTodosLoaded, setAreTodosLoaded] = useState(false);
+  const [isModalShown, setIsModalShown] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState('');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setAreTodosLoaded(false);
+    getTodos().then(todosFromServer => {
+      setAllTodos(todosFromServer);
+      setAreTodosLoaded(true);
+    });
```

Comment:
```suggestion
    setIsLoading(true);
    getTodos().then(setAllTodos).finally(() => setIsLoading(false));
```
You should remove loader in any case, even an error occurs. So it's better to name the state as `isLoading`